tools/clang/include/clang/Basic/DiagnosticParseKinds.td

```diff
@@ -1027,6 +1027,8 @@ def warn_hlsl_semantic_identifier_collision : Warning <
   InGroup< HLSLSemanticIdentifierCollision >;
 def err_hlsl_enum : Error<
   "enum is unsupported in HLSL before 2017">;
+def warn_hlsl_new_feature : Warning <
+  "%0 may not be supported by compilers prior to %1.">;
```

Comment:
being a bit nicpicky, maybe the warning message could be something like:

%0 is an HLSL %1 feature, and is available in older versions as a non-portable extension